### ISSUE TITLE
Target parameter is resolved from environment variable before command-line positional argument

### ISSUE THREAD

**Issue by RobSmyth** [nuke-build/nuke#300]
Hi,

My Windows development environment has a system environment variable "Target". It seems that Nuke keeps picking this up in preference to the command line as I get an error that target "XX" (value of the environment variable) does not exist.

I'm running: `nuke --target Compile`

(I'm a newbie, so maybe I'm just getting the command line wrong?)

It does the same when I invoke via the PowerShell script build.ps1 without arguments.

**Comment by matkoch** [comment]
Hi @RobSmyth,

NUKE picks up parameters (including `ExecutingTargets`) from the command-line / environment variables. You're only chance is to unset the environment  variable inside the build.ps1/sh scripts.

**Comment by RobSmyth** [comment]
Hi @matkoch 

I appreciate the frank response. Thank you.

**Comment by RobSmyth** [comment]
Hi @matkoch ,

Unfortunately, in order to use Nuke, I need to fork and compile a version of nuke with the environment variable "TARGET" renamed to something like "NUKE_TARGET". But looking at the source code I do not see where it is reading the environment variable "TARGET".

Can you help me find the code?

**Comment by matkoch** [comment]
I assume you called `nuke compile` rather than `nuke --target compile`. The latter will work, the first is broken, unfortunately. In general, parameter resolution will first try to resolve from command-line arguments, and then from environment variables.

**Comment by RobSmyth** [comment]
Hi matkoch,

Sorry I have not given this the attention I should have. But coming back to this issue it looks like you have redefined my issue as being about command line argument having precedence over an environment variable and that has been fixed. Thank you.

The fix done is good but it is a work around and does not fix the issue I reported. We are building by a command "nuke" (without parameters). That fails and I assume it will still fail as Nuke and another application are both using an environment variable "TARGET".

I put that both apps have a bug in that neither should be using such a generic environment variable name.

So, while thank you for your providing a work around, I think the reported issue remains. What is the best way to deal with that? Rename this issue or I can create a new one?
